UWP target (e.g. x86_64-uwp-windows-msvc) is broken

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried to compile x86_64-uwp-windows-msvc on nightly rust using `-Z build-std` . I used `staticlib` as target type. Then I tried to link the successfully created .lib file into my exe - outside of rust ecosystem. The linker failed with this error:

```error LNK2019: unresolved external symbol __imp_NtOpenFile referenced in function _ZN3std3sys3pal7windows2fs14remove_dir_all12nt_open_file17h43bb929b19188a87E```

My successful cargo command which created my staticlib which is unusable in UWP is:

```
cargo.exe rustc --lib --target=x86_64-uwp-windows-msvc --package pkg_name --manifest-path C:/path/to/Cargo.toml -Z build-std=std,panic_abort -- -Cdefault-linker-libraries=no
```

The issue is due to rust's std lib using NtOpenFile function which is not available for UWP applications. This regression was introduced in https://github.com/rust-lang/rust/commit/bb9d5c4658218f6e1c477d8c3e0224c1d708e42b

The usage of the function is not guarded by the `#[cfg(target_vendor = "uwp")]` feature here: https://github.com/rust-lang/rust/blob/446649d46383734e9e047d072a482720216dd59c/library/std/src/sys/pal/windows/fs/remove_dir_all.rs#L55

The way I understand it, all uwp targets are unusable since the above mentioned commit around September 2024 because the rust's std library uses function not allowed for uwp applications.

--- a/library/std/src/sys/pal/windows/c.rs
+++ b/library/std/src/sys/pal/windows/c.rs
@@ -237,6 +237,17 @@ compat_fn_with_fallback! {
         STATUS_NOT_IMPLEMENTED
     }
     #[cfg(target_vendor = "uwp")]
+    pub fn NtOpenFile(
+        filehandle: *mut HANDLE,
+        desiredaccess: u32,
+        objectattributes: *const OBJECT_ATTRIBUTES,
+        iostatusblock: *mut IO_STATUS_BLOCK,
+        shareaccess: u32,
+        openoptions: u32
+    ) -> NTSTATUS {
+        STATUS_NOT_IMPLEMENTED
+    }
+    #[cfg(target_vendor = "uwp")]
     pub fn NtReadFile(
         filehandle: HANDLE,
         event: HANDLE,

